File: Frontend/Pharmacy/src/app/dtos/medicine-loss-dto.ts
import { MedicinePurchaseDTO } from "./medicine-purchase-dto";

export interface MedicineLossDTO {
  medicineLossId: number; // Updated
  medicinePurchaseId: number; // Updated
  lossDate: Date;
  quantityLoss: number;
  lossAmount: number;
  totalLoss: number;
  lossReason: string;
  medicinePurchase?: MedicinePurchaseDTO;
}

export interface CreateMedicineLossDTO {
  medicinePurchaseId: number; // Updated
  lossDate: Date;
  quantityLoss: number;
  lossAmount: number;
  totalLoss: number;
  lossReason: string;
}

export interface UpdateMedicineLossDTO {
  medicineLossId: number; // Updated
  medicinePurchaseId: number; // Updated
  lossDate: Date;
  quantityLoss: number;
  lossAmount: number;
  totalLoss: number;
  lossReason: string;
}

File: Backend/Final_Project/Final_Project/DTOs/MedicineLossDTO.cs
namespace Final_Project.DTOs
{
    public class MedicineLossDTO
    {
        public int MedicineLossId { get; set; }
        public int MedicinePurchaseId { get; set; }
        public DateTime LossDate { get; set; }
        public int QuantityLoss { get; set; }
        public decimal LossAmount { get; set; }
        public decimal TotalLoss { get; set; }
        public string LossReason { get; set; }
        public MedicinePurchaseDTO? MedicinePurchase { get; set; }
    }

    public class CreateMedicineLossDTO
    {
        public int MedicinePurchaseId { get; set; }
        public DateTime LossDate { get; set; }
        public int QuantityLoss { get; set; }
        public decimal LossAmount { get; set; }
        public decimal TotalLoss { get; set; }
        public string LossReason { get; set; }
    }

    public class UpdateMedicineLossDTO
    {
        public int MedicineLossId { get; set; }
        public int MedicinePurchaseId { get; set; }
        public DateTime LossDate { get; set; }
        public int QuantityLoss { get; set; }
        public decimal LossAmount { get; set; }
        public decimal TotalLoss { get; set; }
        public string LossReason { get; set; }
    }
}
